other/clang-format-configuration/.clang-format

```diff
@@ -0,0 +1,145 @@
+BasedOnStyle: LLVM
+AccessModifierOffset: -2
+AlignAfterOpenBracket: Align
+AlignConsecutiveAssignments: false
+AlignConsecutiveBitFields: false
+AlignConsecutiveDeclarations: false
+AlignConsecutiveMacros: false
+AlignEscapedNewlines: DontAlign
+AlignOperands: Align
+AlignTrailingComments: true
+AllowAllArgumentsOnNextLine: true
+AllowAllConstructorInitializersOnNextLine: true
+AllowAllParametersOfDeclarationOnNextLine: true
+AllowShortBlocksOnASingleLine: Always
+AllowShortCaseLabelsOnASingleLine: true
+AllowShortEnumsOnASingleLine: true
+AllowShortFunctionsOnASingleLine: Empty
+AllowShortIfStatementsOnASingleLine: Always
+AllowShortLambdasOnASingleLine: Empty
+AllowShortLoopsOnASingleLine: true
+AlwaysBreakAfterDefinitionReturnType: None
+AlwaysBreakAfterReturnType: None
+AlwaysBreakBeforeMultilineStrings: false
+AlwaysBreakTemplateDeclarations: No
+BinPackArguments: true
+BinPackParameters: true
+# Only used when "BreakBeforeBraces" set to "Custom"
+BraceWrapping:
+  AfterCaseLabel: false
+  AfterClass: false
+  AfterControlStatement: Never
+  AfterEnum: false
+  AfterFunction: false
+  AfterNamespace: false
+  #AfterObjCDeclaration:
+  AfterStruct: false
+  AfterUnion: false
+  AfterExternBlock: false
+  BeforeCatch: false
+  BeforeElse: false
+  BeforeLambdaBody: false
+  BeforeWhile: false
+  IndentBraces: false
+  SplitEmptyFunction: false
+  SplitEmptyRecord: false
+  SplitEmptyNamespace: false
+# Java-specific
+#BreakAfterJavaFieldAnnotations:
+BreakBeforeBinaryOperators: NonAssignment
+BreakBeforeBraces: Attach
+BreakBeforeTernaryOperators: true
+BreakConstructorInitializers: BeforeColon
+BreakInheritanceList: BeforeColon
+BreakStringLiterals: false
+ColumnLimit: 0
+# "" matches none
+CommentPragmas: ""
+CompactNamespaces: false
+ConstructorInitializerAllOnOneLineOrOnePerLine: true
+ConstructorInitializerIndentWidth: 2
+ContinuationIndentWidth: 2
+Cpp11BracedListStyle: false
+DeriveLineEnding: true
+DerivePointerAlignment: true
+DisableFormat: false
+# Docs say "Do not use this in config files". The default (LLVM 11.0.1) is "false".
+#ExperimentalAutoDetectBinPacking:
+FixNamespaceComments: false
+ForEachMacros: []
+IncludeBlocks: Preserve
+IncludeCategories: []
+# "" matches none
+IncludeIsMainRegex: ""
+IncludeIsMainSourceRegex: ""
+IndentCaseBlocks: true
+IndentCaseLabels: true
+IndentExternBlock: Indent
+IndentGotoLabels: false
+IndentPPDirectives: None
+IndentWidth: 2
+IndentWrappedFunctionNames: false
+InsertTrailingCommas: None
+# Java-specific
+#JavaImportGroups:
+# JavaScript-specific
+#JavaScriptQuotes:
+#JavaScriptWrapImports
+KeepEmptyLinesAtTheStartOfBlocks: true
+MacroBlockBegin: ""
+MacroBlockEnd: ""
+# Set to a large number to effectively disable
+MaxEmptyLinesToKeep: 100000
+NamespaceIndentation: None
+NamespaceMacros: []
+# Objective C-specific
+#ObjCBinPackProtocolList:
+#ObjCBlockIndentWidth:
+#ObjCBreakBeforeNestedBlockParam:
+#ObjCSpaceAfterProperty:
+#ObjCSpaceBeforeProtocolList
```

Comment:
Are these required?

Reply:
They are irrelevant because the configuration is intended for use with Arduino language/C++/C files.

However, the keys must still be kept in the configuration file because the new system for validating the configuration data is:

1. Run `clang-format --dump-config --style=file` to dump the effective ClangFormat configuration
2. Diff the dumped configuration against the configuration file

The `clang-format --dump-config` output contains these keys (set to their default values) even if the active configuration file does not, so if we remove them from the configuration file then the diff will always fail.

I added notes about all these sorts of keys that are specific to irrelevant languages to reduce the likelihood of them causing confusion in the future:

https://github.com/arduino/tooling-project-assets/blob/c5f256f2aea0b2636a7ff56e450da259d2ca6efe/other/clang-format-configuration/notes.md?plain=1#L99-L123

Note that they are not commented in the final configuration file:

https://github.com/arduino/tooling-project-assets/blob/c5f256f2aea0b2636a7ff56e450da259d2ca6efe/other/clang-format-configuration/.clang-format#L101-L105

(these are the default values from ClangFormat)

Reply:
I've seen noticed this when I continued with the review, thanks Per